src/editor/Core/Reducer.re

```diff
@@ -67,6 +73,12 @@ let reduce: (State.t, Actions.t) => State.t =
     | ChangeMode(m) =>
       let ret: State.t = {...s, mode: m};
       ret;
+    | BufferWritePost(bs) => {
+        ...s,
+        activeBufferId: bs.bufferId,
```

Comment:
Is it possible to save a buffer if it isn't the active one?

Reply:
you could `writeall`, but you cant target an inactive buffer and save it, or rather I'm pretty sure that's not possible. But this repopulates the bufferlist so we'd find out anyway?

Reply:
actually you can explicitly call `writeall` its only available as `autowriteall` a config option you set in your vimrc

Reply:
Hmm, I was curious why we need to set the `activeBufferId` here - it's possible for the active buffer to change when we're writing it?

Reply:
I was thinking of cases where if you run an `argadd` command then `bufdo` you can essentially create add a bunch of buffers as arguments then run `bufdo` command and example would be if you run a regex on multiple files at a time, not sure what the autocommand flow would be in those kind of cases. It's possibly/probably unnecessary but we

* have the information as part of the `AutoCommandContext`
* were already setting state
* could catch edge cases

Reply:
Sounds good! Thanks for the info.